src/Gemstone.Timeseries/Model/MeasurementExpressionParser.cs

```diff
@@ -0,0 +1,317 @@
+            if (signalTypeAcronym.Equals("ALOG", StringComparison.OrdinalIgnoreCase) || signalTypeAcronym.Equals("DIGI", StringComparison.OrdinalIgnoreCase))
+                return GuessBaseKV(null, deviceAcronym, "CALC");
+        }
+
+        foreach (string voltageLevel in s_commonVoltageLevels)
+        {
+            if (deviceAcronym.IndexOf(voltageLevel, StringComparison.Ordinal) > -1)
+                return int.Parse(voltageLevel);
+        }
+
+        return 0;
+    }
+
+    #endregion
+
+    #region [ Static ]
+
+    private static Dictionary<string, DataRow>? s_signalTypes;
+    private static Dictionary<string, DataRow>? s_companies;
+    private static Dictionary<string, DataRow>? s_interconnections;
+    private static Dictionary<string, DataRow>? s_Vendors;
+
+
+    private static readonly string[] s_commonVoltageLevels = CommonVoltageLevels.Values;
+
+    private static Dictionary<string, DataRow> InitializeSignalTypes()
+    {
+        // It is expected that when a point tag is needing to be created that the database will be available
+        using AdoDataConnection database = new(ConfigSettings.Default);
+        Dictionary<string, DataRow> signalTypes = new(StringComparer.OrdinalIgnoreCase);
+
+        foreach (DataRow row in database.Connection.RetrieveData("SELECT * FROM SignalType").AsEnumerable())
+        {
+            if (row is null)
+                continue;
+
+            signalTypes.AddOrUpdate(row["Acronym"]?.ToString() ?? "", row);
+        }
+
+        return signalTypes;
+    }
+
+    private static Dictionary<string, DataRow> InitializeCompanies()
+    {
+        // It is expected that when a point tag is needing to be created that the database will be available
+        using AdoDataConnection database = new(ConfigSettings.Default);
+        Dictionary<string, DataRow> companies = new(StringComparer.OrdinalIgnoreCase);
+
+        foreach (DataRow row in database.Connection.RetrieveData("SELECT * FROM Company").AsEnumerable())
+        {
+            if (row is null)
+                continue;
+
+            companies.AddOrUpdate(row["Acronym"]?.ToString() ?? "", row);
+        }
+
+        return companies;
+    }
+
+    private static Dictionary<string, DataRow> InitializeInterconnections()
+    {
+        // It is expected that when a point tag is needing to be created that the database will be available
+        using AdoDataConnection database = new(ConfigSettings.Default);
+        Dictionary<string, DataRow> interconnections = new(StringComparer.OrdinalIgnoreCase);
+
+        foreach (DataRow row in database.Connection.RetrieveData("SELECT * FROM Interconnection").AsEnumerable())
+        {
+            if (row is null)
+                continue;
+
+            interconnections.AddOrUpdate(row["Acronym"]?.ToString() ?? "", row);
+        }
+
+        return interconnections;
+    }
+
+    private static Dictionary<string, DataRow> IntializeVendor()
```

Comment:
Rename to make plural:
```suggestion
    private static Dictionary<string, DataRow> IntializeVendors()
```
Then fix reference.